### ISSUE TITLE
NullPointerException

### ISSUE THREAD

**Issue by zhancheng** [CarnegieTechnologies/gallery_saver#120]
help wanted, many users report this crash info, full stack here
```

# main(2)

kotlin.KotlinNullPointerException

carnegietechnologies.gallery_saver.GallerySaver.void finishWithSuccess()(SourceFile:1)
carnegietechnologies.gallery_saver.GallerySaver.void access$finishWithSuccess(carnegietechnologies.gallery_saver.GallerySaver)(SourceFile:1)
carnegietechnologies.gallery_saver.GallerySaver$saveMediaFile$1.java.lang.Object invokeSuspend(java.lang.Object)(SourceFile:6)
kotlin.coroutines.jvm.internal.BaseContinuationImpl.void resumeWith(java.lang.Object)(SourceFile:3)
kotlinx.coroutines.DispatchedTask.void run()(SourceFile:18)
android.os.Handler.handleCallback(Handler.java:790)
android.os.Handler.dispatchMessage(Handler.java:99)
android.os.Looper.loop(Looper.java:192)
android.app.ActivityThread.main(ActivityThread.java:6770)
java.lang.reflect.Method.invoke(Native Method)
com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:556)
com.android.internal.os.ZygoteInit.main(ZygoteInit.java:875)
```

**Comment by jamesdixon** [comment]
@zhancheng 

I had this happen as well but it only happened in the case where I was trying to save more than 1 photo at a time. For example, my camera allows a user to take multiple photos and then tap a button to save them. When I attempted to save more than 1 photo to the gallery, I received this error.

I have `GallerySaver.saveImage` as part of a larger method:

```
  /// Saves a photo to the gallery.
  Future<bool> savePhotoToGallery({
    @required String filePath,
    @required String propertyAddress,
    bool forceSave = false,
    bool isEdited = false,
  }) async {
    final settings = ref.read(settingsServiceProvider).get();

    final shouldSaveOriginals =
        (!isEdited && settings.saveMediaToGallery) || forceSave;
    final shouldSaveAnnotated =
        isEdited && settings.saveMediaToGallery && settings.saveAnnotatedImages;

    if (shouldSaveOriginals || shouldSaveAnnotated) {
      final currentDate = Jiffy().format('yyyy-MM-dd');
      final albumName = '$currentDate - $propertyAddress';

      return GallerySaver.saveImage(filePath, albumName: albumName);
    }

    return false;
  }
```

I then call it like so through this method:

```
  void _onCameraIconPress({
    @required BuildContext context,
    @required String propertyAddress,
  }) async {
    assert(context != null);
    assert(propertyAddress.isPresent);
    final mediaService = context.read(mediaServiceProvider);
    final mediaStorageService = context.read(mediaStorageServiceProvider);

    final photos = await mediaService.takeMultiplePictures();

    for (final photo in photos) {
      // ignore: unawaited_futures
      mediaStorageService.savePhotoToGallery(
        filePath: photo.path,
        propertyAddress: propertyAddress,
        forceSave: true,
      );
    }
  }
```

You'll notice that when I call the `mediaStorageService.savePhotoToGallery`, I wasn't `await`ing the Future as I just wanted to process as fast as possible. 

The fix for me was to `await` that Future.

I hope this helps!

**Comment by lecicdjuro** [comment]
solved with 2.2.0
